### ISSUE TITLE
[FEATURE] Support loading multiple documents from an S3 compatible object storage bucket

### ISSUE THREAD

**Issue by slaplante-raft** [FlowiseAI/Flowise#2778]
**Describe the feature you'd like**
Support loading multiple documents from an S3 compatible object storage bucket (ex MinIO). We found that currently only a single file from an s3 bucket can be loaded using the current s3 document loader component. In addition to that, we want to load documents from a MinIO bucket which requires a URL to point to our storage server. However, the current component doesn't have this option. 

**Additional context**
Ive implemented my own component which lists the contents of a bucket(using ListObjectsV2Command), takes the output from that and issues multiple GetObjectCommands to download the files to a local filesystem. Then i used the LangChain.js [DirectoryLoader](https://v02.api.js.langchain.com/classes/langchain_document_loaders_fs_directory.DirectoryLoader.html) to load the documents from the local directory. Id like to contribute this back to the community. However, i want to make sure the method i choose is acceptable and in line with what the contributors think. 

Questions:
- The current S3File.ts component only allows one file to be loaded. Was this a deliberate choice so a user cant shoot themselves in the foot?
- An [Unstructured](https://docs.unstructured.io/) server is required to load documents for the current S3 component. Was there a reason the DirectoryLoader cant be used?
- Would it be better to create an S3DirectoryLoader in LangChain.js like the [python](https://python.langchain.com/v0.2/docs/integrations/document_loaders/aws_s3_directory/) version has? Then i can create an S3Directory.ts component in Flowise that uses the new LangChain.js S3DirectoryLoader

**Comment by HenryHengZJ** [comment]
Appreciated the effort for creating the component! @slaplante-raft 

TLDR: yes you can create a S3 Directory Loader, and that would be a bettter solution instead of changing the S3File.

1.) The current S3File is a design choice, by allowing users to download a single file, store it into a `tmp` file path, pass that to Unstructured.
2.) No, I was also thinking of allowing user to select directoryloader or unstructured
3.) answered above :)

**Comment by slaplante-raft** [comment]
Thanks for the reply @HenryHengZJ I just opened this [pull request](https://github.com/FlowiseAI/Flowise/pull/2818) which introduces an S3 Directory Loader component. Looking forward to some feedback!
